File: collect_app/src/androidTest/java/org/odk/collect/android/support/pages/ErrorDialog.java
package org.odk.collect.android.support.pages;

public class ErrorDialog extends OkDialog {

    @Override
    public ErrorDialog assertOnPage() {
        super.assertOnPage();
        assertText(org.odk.collect.strings.R.string.error_occured);
        return this;
    }
}

File: collect_app/src/androidTest/java/org/odk/collect/android/support/pages/ErrorDialog.kt
package org.odk.collect.android.support.pages

import org.odk.collect.strings.R

class ErrorDialog : OkDialog() {
    fun assertOnPage(isFatal: Boolean): ErrorDialog {
        assertOnPage()
        if (isFatal) {
            assertText(R.string.form_cannot_be_used)
        } else {
            assertText(R.string.error_occured)
        }
        return this
    }
}
